### PR TITLE
Improve ArduinoJson and add 6.21.2

### PR DESCRIPTION
**Pull request by jputcu**
Specify library name and version:  **arduinojson/6.21.2**

Fix blocking build issue on conan1.
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/arduinojson/all/conanfile.py [hunk 1]
```diff
@@ -16,12 +17,29 @@ class ArduinojsonConan(ConanFile):
...
         if Version(self.version) >= "6.21.0":
-            check_min_cppstd(self, 11)
+            if self.settings.compiler.get_safe("cppstd"):
+                check_min_cppstd(self, 11)
+            elif str(self.settings.compiler) == 'apple-clang':
+                raise ConanInvalidConfiguration("cppstd needs to be set on apple-clang to activate c++11.")
+            else:
+                minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+                if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                    raise ConanInvalidConfiguration("{self.ref} requires C++11, which your compiler does not support.")
```
Comment (prince-chrismc) [inline recipes/arduinojson/all/conanfile.py:42]:
```suggestion
                    raise ConanInvalidConfiguration(f"{self.ref} requires C++11, which your compiler does not support.")
```

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/arduinojson/all/conanfile.py [hunk 2]
```diff
@@ -16,12 +17,29 @@ class ArduinojsonConan(ConanFile):
...
+            "clang": "6",
+        }
+
     def validate(self):
         if Version(self.version) >= "6.21.0":
-            check_min_cppstd(self, 11)
+            if self.settings.compiler.get_safe("cppstd"):
+                check_min_cppstd(self, 11)
+            elif str(self.settings.compiler) == 'apple-clang':
+                raise ConanInvalidConfiguration("cppstd needs to be set on apple-clang to activate c++11.")
```
Comment (SpaceIm) [inline recipes/arduinojson/all/conanfile.py:38]: That's not a good recipe design, remove this please and use cxx_std_11 in test package.

File: recipes/arduinojson/all/conanfile.py [hunk 3]
```diff
@@ -16,12 +17,29 @@ class ArduinojsonConan(ConanFile):
...
     url = "https://github.com/conan-io/conan-center-index"
+    package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
```
Comment (SpaceIm) [inline recipes/arduinojson/all/conanfile.py:28]: Are you sure? Some C++11 features would require at least Visual Studio 2019 here?

File: recipes/arduinojson/all/conanfile.py (hunk 2 above)
Comment (jputcu) [reply recipes/arduinojson/all/conanfile.py:38]: Will do, couldn't wrap my head around getting the CI to build for C++11, but if it is just trying to compile the example I can indeed force C++11 when the ArduinoJson version is >=6.21.

File: recipes/arduinojson/all/conanfile.py (hunk 3 above)
Comment (jputcu) [reply recipes/arduinojson/all/conanfile.py:28]: This was a quick workaround to get the C++11, but your pull request has nicer way.

**Pull request merged**